Enable valgrind for glibc 2.19 targets, and update the error from 2.17
to 2.19 (which in reality was 2.18).
Status: upstream.

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

diff -Nura valgrind-3.9.0.orig/configure.ac valgrind-3.9.0/configure.ac
--- valgrind-3.9.0.orig/configure.ac	2014-06-18 19:06:39.212571710 -0300
+++ valgrind-3.9.0/configure.ac	2014-06-18 19:08:37.385557736 -0300
@@ -918,6 +918,13 @@
 	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
 	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
 	;;
+     2.19)
+        AC_MSG_RESULT(2.19 family)
+	AC_DEFINE([GLIBC_2_19], 1, [Define to 1 if you're using glibc 2.19.x])
+	DEFAULT_SUPP="glibc-2.X.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
+	;;
      darwin)
 	AC_MSG_RESULT(Darwin)
 	AC_DEFINE([DARWIN_LIBC], 1, [Define to 1 if you're using Darwin])
@@ -931,7 +938,7 @@
 
      *)
 	AC_MSG_RESULT([unsupported version ${GLIBC_VERSION}])
-	AC_MSG_ERROR([Valgrind requires glibc version 2.2 - 2.17])
+	AC_MSG_ERROR([Valgrind requires glibc version 2.2 - 2.19])
 	AC_MSG_ERROR([or Darwin libc])
 	;;
 esac
